### ISSUE TITLE
Segfault when playing a specific instrument

### ISSUE THREAD

**Issue by gera2342** [schismtracker/schismtracker#499]
Module in question: https://modarchive.org/index.php?request=view_by_moduleid&query=207260

Play note F-6 with instrument 20 or wait for pattern 10 (position 11)

```gdb
Thread 10 "SDLAudioP2" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffceffe6c0 (LWP 13885)]
0x000055555558396f in FastMono8BitMix (channel=0x7fffe0382018, pbuffer=<optimized out>, pbufmax=0x7fffe037f010) at player/mixer.c:475
475	DEFINE_MIX_INTERFACE_FAST(8)
(gdb) bt
#0  0x000055555558396f in FastMono8BitMix (channel=0x7fffe0382018, pbuffer=<optimized out>, pbufmax=0x7fffe037f010) at player/mixer.c:475
#1  0x000055555558afa6 in csf_create_stereo_mix (csf=csf@entry=0x7fffe037e010, count=count@entry=512) at player/mixer.c:837
#2  0x000055555558f697 in csf_read (csf=0x7fffe037e010, v_buffer=v_buffer@entry=0x5555564b88b0, bufsize=bufsize@entry=4096) at player/sndmix.c:824
#3  0x0000555555593564 in audio_callback (qq=<optimized out>, stream=0x5555564b88b0 "", len=4096) at schism/audio_playback.c:142
```

**Comment by brandflake11** [comment]
It doesn't crash for me on both the release 20240630 or the current git snapshot as of this time. What are your system specs (operating system, processor architecture, version of schism tracker)? (Also note, I'm not a developer of schism, just a user).

I'm using an Arch Linux container with distrobox and compiled schismtracker myself from the source release and the git snapshot.

**Comment by gera2342** [comment]
Gentoo, Pipewire, amd64, gcc compiler, latest master

I tried also building on windows, the 20240630 tag does not crash, master does

Tag 20240630 also works on Linux

**Comment by gera2342** [comment]
Ok I did git-bisect which led me to this commit 209e93e0c2beddf92f257aa6411f09084071c8a4
@brandflake11 does it crash for you on this commit?

**Comment by mrpapersonic** [comment]
yep, that commit caused something to break since I've been getting intermittent crashes because of it too.

Which interpolation method are you using?

**Comment by gera2342** [comment]
non-interpolated and linear crash
cubic spline and 8-tap fir filter do not crash

**Comment by mrpapersonic** [comment]
fixed with 05570ffe877637f04665fc1dd050f6a99feecf8d
